core/src/test/scala/org/http4s/TestInstances.scala

```diff
@@ -191,6 +191,27 @@ trait TestInstances {
         arbitrary[Header.Raw]
       )
     }
+
+  implicit lazy val arbitraryServerSentEvent: Arbitrary[ServerSentEvent] = {
+    def singleLineString: Gen[String] =
+      arbitrary[String] suchThat { s => !s.contains("\r") && !s.contains("\n") }
+    Arbitrary(for {
+      data <- singleLineString
+      event <- frequency(
+        4 -> None,
+        1 -> singleLineString.map(Some.apply)
```

Comment:
Nothing in the spec permits a \n or \r in the event.  How do we want to prevent their construction?  Likewise for id.